Jenkinsfile

```diff
@@ -1463,7 +1484,44 @@ pipeline {
                                     failure_artifacts: env.STAGE_NAME, ignore_failure: true
                         }
                     }
-                }
+                } // stage('Test CentOS 7 RPMs')
+                stage('Scan CentOS 7 RPMs') {
+                    when {
+                        beforeAgent true
+                        allOf {
+                            not { branch 'weekly-testing' }
+                            not { environment name: 'CHANGE_TARGET', value: 'weekly-testing' }
+                            // expression { ! skip_stage('scan-centos-rpms') }
+                        }
+                    }
+                    agent {
+                        label 'ci_vm1'
+                    }
+                    steps {
+                        unstash 'CentOS-rpm-version'
+                        script {
+                            daos_packages_version = readFile('centos7-rpm-version').trim()
+                        }
+                        provisionNodes NODELIST: env.NODELIST,
+                                       node_count: 1,
+                                       profile: 'daos_ci',
+                                       distro: 'el7',
+                                       snapshot: true,
+                                       inst_repos: el7_daos_repos,
+                                       inst_rpms: 'environment-modules ' +
+                                                  'clamav clamav-devel'
+                        catchError(stageResult: 'UNSTABLE', buildResult: 'SUCCESS') {
+                            runTest script: "${rpm_test_pre}" +
+                                            "sudo yum -y install daos-client-${daos_packages_version}\n" +
+                                            "sudo yum -y history rollback last-1\n" +
+                                            "sudo yum -y install daos-server-${daos_packages_version}\n" +
+                                            "sudo yum -y install daos-tests-${daos_packages_version}\n" +
+                                            "${rpm_scan_daos_test}" + '"',
+                                    junit_files: null,
```

Comment:
Can we return a junit result instead of having people have to crawl this log if it fails?

All of our tests should be striving for junit output.  We're a ways away still, but that's the ultimate goal.  We want log trawling to be the exception.

Reply:
This is a binary pass/fail test.
If we actually find any alleged malware, we will have to look over a lot of things to see how that happened.
Does not seem worth generating a single junit file just for this right now.  Maybe in the future.

Reply:
But surely there is a list of infected files if any are found, etc. yes?

That's the kind of information that should be in a JUnit result.

Reply:
I have not found a way to get the list of files in a format that is easily captured and used, and did not investigate it that much since it was not specifically required for this milestone.

Reply:
Reporting the result in the manner preferred by the tool should be part of the task/milestone.

Wouldn't the list of files be in `/tmp/clamscan.out` where you are already doing the `grep 'Infected files: 0$'` from?  Even if you just dumped that whole file into a single JUnit "blob" result as a failure would be better than not producing any JUnit result at all.

Ultimately making people trudge through processing logs is hurting the reputation of our CI by making failures difficult to find and understand.  The ultimate goal to resolving that is to have **all** test results in one place, which is fed by the various JUnit outputs.

If the stage runs successfully, (even if it finds failures) it should exit 0.  Those failures should be reflected in the JUnit "Test Result" that Jenkins provides.

Completely agreed that we have lots of places where this is not (yet) true.  But I'd like us not to take a step backwards towards the goal of making CI easier and more consistent to use when avoiding the step backwards is not terribly difficult.

Reply:
It is now generating a Junit test report of pass/fail.  On failure it should capture the log.